test/e2e/quick_start_test.go

```diff
@@ -96,27 +96,45 @@ var _ = Describe("Quick start", func() {
 												)
 											}
 
-											// For Nutanix provider, reserve an IP address for the workload cluster control plane endpoint -
-											// remember to unreserve it!
+											// For Nutanix provider, reserve an IP address for the workload cluster:
+											// 1. control plane endpoint
+											// 2. service load balancer
+											// Remember to unreserve it after the test!
 											if provider == "Nutanix" {
-												By(
-													"Reserving an IP address for the workload cluster control plane endpoint",
-												)
 												nutanixClient, err := nutanix.NewV4Client(
 													nutanix.CredentialsFromCAPIE2EConfig(testE2EConfig),
 												)
 												Expect(err).ToNot(HaveOccurred())
+												subnetName := testE2EConfig.MustGetVariable("NUTANIX_SUBNET_NAME")
+												prismElementClusterName := testE2EConfig.MustGetVariable(
+													"NUTANIX_PRISM_ELEMENT_CLUSTER_NAME",
+												)
 
+												By(
+													"Reserving an IP address for the workload cluster control plane endpoint",
+												)
 												controlPlaneEndpointIP, unreserveControlPlaneEndpointIP, err := nutanix.ReserveIP(
+													subnetName,
+													prismElementClusterName,
+													nutanixClient,
+												)
+												Expect(err).ToNot(HaveOccurred())
+												DeferCleanup(unreserveControlPlaneEndpointIP)
+												testE2EConfig.Variables["CONTROL_PLANE_ENDPOINT_IP"] = controlPlaneEndpointIP
+
+												By(
+													"Reserving an IP address for the workload cluster kubernetes Service load balancer",
+												)
+												kubernetesServiceLoadBalancerIP, unreservekubernetesServiceLoadBalancerIP, err := nutanix.ReserveIP(
 													testE2EConfig.MustGetVariable("NUTANIX_SUBNET_NAME"),
 													testE2EConfig.MustGetVariable(
 														"NUTANIX_PRISM_ELEMENT_CLUSTER_NAME",
```

Comment:
Why duplicate the func calls testE2EConfig.MustGetVariable() here, and not just use the local variable "subnetName" and "prismElementClusterName" holding the expected values?

Reply:
Thanks fixing, bad rebase